test/Microsoft.Framework.Primitives.Tests/StringValuesTests.cs

```diff
@@ -23,51 +90,34 @@ public void IsReadOnly_True()
             Assert.Throws<NotSupportedException>(() => ((ICollection<string>)stringValues).Clear());
         }
 
-        [Fact]
-        public void DefaultConstructor_ExpectedValues()
+        [Theory]
+        [MemberData(nameof(DefaultOrNullStringValues))]
+        public void DefaultOrNull_ExpectedValues(StringValues stringValues)
         {
-            var stringValues = new StringValues();
-            Assert.Equal(0, stringValues.Count);
-            Assert.Equal((string)null, stringValues);
-            Assert.Equal(new string[0], stringValues.ToArray());
-
-            Assert.True(StringValues.IsNullOrEmpty(stringValues));
-            Assert.Throws<IndexOutOfRangeException>(() => stringValues[0]);
-            Assert.Equal(string.Empty, stringValues.ToString());
-            Assert.Equal(-1, ((IList<string>)stringValues).IndexOf(string.Empty));
-            Assert.Equal(0, stringValues.Count());
-        }
-
-        [Fact]
-        public void Constructor_NullStringValue_ExpectedValues()
-        {
-            var stringValues = new StringValues((string)null);
-            Assert.Equal(0, stringValues.Count);
-            Assert.Null((string)stringValues);
-            Assert.Equal(string.Empty, stringValues.ToString());
             Assert.Null((string[])stringValues);
-            Assert.Equal(new string[0], stringValues.ToArray());
-
-            Assert.True(StringValues.IsNullOrEmpty(stringValues));
-            Assert.Throws<IndexOutOfRangeException>(() => stringValues[0]);
-            Assert.Equal(-1, ((IList<string>)stringValues).IndexOf(string.Empty));
-            Assert.Equal(0, stringValues.Count());
         }
 
-        [Fact]
-        public void Constructor_NullStringArray_ExpectedValues()
+        [Theory]
+        [MemberData(nameof(DefaultOrNullStringValues))]
+        [MemberData(nameof(EmptyStringValues))]
+        public void DefaultNullOrEmpty_ExpectedValues(StringValues stringValues)
         {
-            var stringValues = new StringValues((string[])null);
             Assert.Equal(0, stringValues.Count);
             Assert.Null((string)stringValues);
+            Assert.Equal((string)null, stringValues);
```

Comment:
Won't StringValues(string.Empty) implicitly convert to string.Empty here? How is null == string.Empty?